src/ctapipe/calib/camera/calibrator.py

```diff
@@ -292,13 +311,21 @@ def _calibrate_dl1(self, event, tel_id):
             and dl1_calib.absolute_factor is not None
         ):
             if selected_gain_channel is None:
-                dl1.image *= dl1_calib.relative_factor / dl1_calib.absolute_factor
+                if self.image_calibration_type == "charge":
+                    dl1.image *= dl1_calib.relative_factor / dl1_calib.absolute_factor
+                elif self.image_calibration_type == "variance":
+                    dl1.image *= np.sqrt(
+                        dl1_calib.relative_factor / dl1_calib.absolute_factor
+                    )
```

Comment:
You could check here if the `extractor.__class__.__name__` is `VarianceExtractor` rather than passing the specific config parameter.